js/foundation.util.core.js

```diff
@@ -61,4 +61,18 @@ function transitionend($elem){
   }
 }
 
-export {rtl, GetYoDigits, RegExpEscape, transitionend};
+/**
+ * Call the given function once the window is loaded,
+ * or immediately if the window is already loaded.
+ * @function
+ *
+ * @param {Function} fn - function to call on window load.
+ */
+function onLoad(fn) {
+  if (document.readyState === 'complete')
+    setTimeout(() => fn(), 0);
```

Comment:
This should be `fn();` as `setTimeout` creates a new context and does not bind `_this` and nother variables in this case.

Reply:
There is no context to preserve. `_this` and other variables are not passed through the context.

Reply:
Take a look at the failing unit test(s) due to this ;-)

Reply:
Right, but this has nothing to do with the context, as `this` is not used.

https://github.com/ncoden/foundation-sites/blob/f353fc2f1a6d556a511734fa7eeacd0221cd064d/js/foundation.offcanvas.js#L169-L173

Reply:
The error comes from the component being already destroyed when the function in `setTimeout` is executed. I don't exactly know why, but the best solution would be to handle this case as we do for others events listeners by unregistering it on `destroy`.